### ISSUE TITLE
train_test_split does not work for Pandas Series with non Dense Index

### ISSUE THREAD

**Issue by cancan101** [dnouri/nolearn#7]
If using a Pandas Series where the Index values are not dense, then `train_test_split` will select index values that do not exist.  

This because [this line](https://github.com/dnouri/nolearn/blob/a9ac3e7310a7208ddf178c95dd7ac9cedf3e8df1/nolearn/lasagne.py#L257):
```
train_indices, valid_indices = iter(kf).next()
```
returns the row numbers but then the indexing is done with index values`

**Comment by dnouri** [comment]
Why wouldn't you reset the index before sending the dataframe in?

**Comment by cancan101** [comment]
That is how I ended up solving the issue. It is worth noting that scikit learn's shuffle / split "just works" without resetting the index using safe indexing (https://github.com/scikit-learn/scikit-learn/blob/f0fe4afd1ee3c19ed13a13f25573f9c173e9e65b/sklearn/utils/__init__.py#L135$

**Comment by dnouri** [comment]
I think we'll be able to resolve this as part of #47.
